packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -186,7 +192,17 @@ export abstract class SharedObject<TEvent extends ISharedObjectEvents = ISharedO
     /**
      * {@inheritDoc (ISharedObject:interface).snapshot}
      */
-    public abstract snapshot(): ITree;
+    public snapshot(): ITree {
+        this.runtime.IFluidSerializer.startRecordingRoutes();
```

Comment:
It might be worth for each SharedObject to have its own FluidSerializer. The serializer can then store the referenced routes per DDS.

Reply:
Another option can be is to have a 'serialize' function in the SharedObject that DDSs call to serialize handles. This 'serialize` method stores the 'path' of the serialized handle. This way we don't have to modify the serializer.

Reply:
Yes for new serialize. It's trivial to make it. I just moved it from container to data store layer, but we can just make them at DDS level. I'd recreated them on demand (for summary) instead of adding some APIs to capture - that would not work before my recent change because I believe we summarize data stores in parallel (so you are getting lucky here with timing :))

Reply:
I'd go even further and explore passing serializer into snapshot() API and down, to be more explicit.
That has a risk though, as code may accidently use instance on data store. So not sure

Reply:
I had moved FluidSerilizer from Container to data store in this change too but your change beat me to it :-).
I like the idea of it being per-DDS level too because its not expensive at all and makes the code simpler.

I am not very comfortable with passing it to snapshot() because of the reasons you mentioned. It increases the surface area of things going wrong.